site/components/pages/Pledge/InvitationModals/AcceptModal/index.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { useState } from "react";
+import { t, Trans } from "@lingui/macro";
+import {
+  ButtonPrimary,
+  ButtonSecondary,
+  Spinner,
+  Text,
+} from "@klimadao/lib/components";
+import { concatAddress, useWeb3 } from "@klimadao/lib/utils";
+
+import { Modal } from "components/Modal";
+import {
+  approveSecondaryWallet,
+  removeSecondaryWallet,
+} from "../../lib/editPledgeMessage";
+import { Pledge } from "../../types";
+import { putPledge, pledgeFormAdapter } from "../../lib";
+import * as styles from "../styles";
+
+type Props = {
+  setShowAcceptModal: (value: boolean) => void;
+  showAcceptModal: boolean;
+  pledge: Pledge;
+  pageAddress: string;
+};
+type Steps = "accept" | "confirm" | "error" | "loading";
+
+export const AcceptModal = (props: Props) => {
+  const [step, setStep] = useState<Steps>("accept");
+  const [errorMessage, setErrorMessage] = useState(null);
+  const shortenedAddress = concatAddress(props.pledge.ownerAddress);
+
+  const getTitle = (step: string) =>
+    ({
+      accept: t({
+        id: "pledge.invitation.accept",
+        message: "Accept Invitation",
+      }),
+      confirm: t({
+        id: "shared.confirm",
+        message: "Confirm",
+      }),
+      error: t({
+        id: "pledge.invitation.error_title",
+        message: "Server Error",
+      }),
+      loading: t({
+        id: "shared.loading",
+        message: "Loading",
+      }),
+    }[step]);
+  const { signer } = useWeb3();
+
+  const handleSubmit = async (params: { message: string }) => {
+    try {
+      if (!signer) return;
+      const address = await signer.getAddress();
+      const signature = await signer.signMessage(params.message);
+      setStep("loading");
+      await putPledge({
+        pageAddress: props.pageAddress,
+        secondaryWalletAddress: address,
+        pledge: pledgeFormAdapter(props.pledge),
+        signature,
+        urlPath: `/pledge/${props.pageAddress}`,
+      });
+      props.setShowAcceptModal(false);
+    } catch (e: any) {
+      console.log("error:", e);
+      setStep("error");
+      setErrorMessage(
+        e.message ??
```

Comment:
do we translate the message thrown by `putPledge` or the server? (I didn't get to that part of the code yet)